compiler/codegen/codegen.cpp

```diff
@@ -2511,7 +2511,17 @@ static void embedGpuCode() {
 }
 
 static void codegenGpuGlobals() {
-  genGlobalInt("chpl_nodeID", 0, false);
+  GenInfo* info = gGenInfo;
+
+  // codegen chpl_nodeID
+  const char* cname = "chpl_nodeID";
```

Comment:
This code is essentially a copy-paste of `genGlobalInt` without the call to `globalInt->setConstant(true);` right?

Rather than copying it, could we add a `setConstant` param to `genGlobalInt` that's `=true` by default but in this case calledwith `=false`.

Given the unexpected/divergent behavior between AMD/NVIDIA with this I'd even go as far as adding an assertion in `genGlobalInt` that this is `==false` when called during our gpu-pass of codegen (we don't call it anywhere else right?). Seems like in the long run we'd want this function to actually put the variable into GPU constant memory but in the absence of that this seems like a way of avoiding duplicating code and making it more likely that we'll fall into this trap again.

Reply:
I agree with that.

This function started as more different than `genGlobalInt` as I was trying different things and didn't want to disrupt other uses of it. In the final version, I still wanted to keep it separate, because I imagine (as you also alluded to) we'll want to handle things more differently for GPUs, and will need a separate utility for them. But how near that is, I don't know. I'd be fine with following your suggestion to reduce code duplication in the meantime.